src/psyclone/psyir/nodes/routine.py

```diff
@@ -202,7 +201,7 @@ def name(self, new_name):
             if 'own_routine_symbol' in self.symbol_table.tags_dict:
                 existing_symbol = self.symbol_table.lookup_with_tag(
                         'own_routine_symbol', scope_limit=self)
-                if existing_symbol.name == new_name:
+                if existing_symbol.name.lower() == new_name.lower():
```

Comment:
The comparison because:
- It maintains the case when regenerating code (*unless it uses different case in different references as we just make 1 per symbol)
- Parsing is just one entry point, we want the comparison to still work if the existing symbol is introduced manually (e.g. in a transformation) and we are not careful with the case.

(that said I would see the argument for forgetting about regenerating original case and just lower once inside the Symbol class constructor, but at the moment these comparisons with lowering and common in the code. Maybe for a different issue? )

Reply:
> The comparison because:
> 
> * It maintains the case when regenerating code (*unless it uses different case in different references as we just make 1 per symbol)

Which does happen in NEMO code. Sometimes they name a function in a different case to its output variable.